dist/js/tether.js

```diff
@@ -1364,34 +1364,32 @@ TetherBase.modules.push({
       }
 
       if (changeAttachY === 'together') {
-        if (top < bounds[1] && tAttachment.top === 'top') {
-          if (eAttachment.top === 'bottom') {
+        if (tAttachment.top === 'top') {
+          if (eAttachment.top === 'bottom' && top < bounds[1]) {
             top += targetHeight;
             tAttachment.top = 'bottom';
 
             top += height;
             eAttachment.top = 'top';
-          } else if (eAttachment.top === 'top') {
-            top += targetHeight;
+          } else if (eAttachment.top === 'top' && top + height > bounds[3] && top - (height - targetHeight) >= bounds[1]) {
```

Comment:
If both attachment and target attachment are "top left" for example, it doesn't make sense to check against `bounds[1]`. Instead, what we should check, is if the bottom of the element surpasses `bounds[3]` and if there is room to show the element without going above `bounds[1]`. The opposite goes for attachments on "bottom".